docs/build_pipeline_controller.md

```diff
@@ -0,0 +1,36 @@
+<div align="center"><h1>BuildPipeline Controller</h1></div>
+
+```mermaid
+
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+%% Defining the styles
+  classDef Red fill:#FF9999;
+  classDef Amber fill:#FFDEAD;
+  classDef Green fill:#BDFFA4;
+ 
+  %% Node definitions
+predicate((PREDICATE: <br> filter events related to <br> PipelineRun <br> that are signed and <br> have succeeded))
+get_pipeline_run{Get the pipeline run<br>from the cluster}
+get_component{Get the component<br>associated with the pipeline run}
+get_application{Get the application<br>associated with the component}
+error[returns error]
+create_adapter[Create adapter and<br>ensure snapshot exists]
```

Comment:
We'll probably need some more details on this logic - outlining at least the following:

- Determine if the Snapshot needs to be created (does it already exist? is this the newest build of that component?)
- Create the Snapshot by getting all existing components and adding/overwriting the newly built one
- Annotate the build pipelineRun with the name of the new Snapshot

Reply:
done